src/txmempool.cpp

```diff
@@ -688,6 +704,23 @@ void CTxMemPool::removeConflicts(const CTransaction &tx, std::list<CTransaction>
             }
         }
     }
+
+    for(const CTxCeasedSidechainWithdrawalInput& csw: tx.GetVcswCcIn()) {
+        if (mapSidechains.count(csw.scId)) {
+            const auto& cswNullifierPos = mapSidechains.at(csw.scId).cswNullifiers.find(csw.nullifier);
+            if(cswNullifierPos != mapSidechains.at(csw.scId).cswNullifiers.end()) {
+                const uint256& txHash = cswNullifierPos->second;
+                const auto& it = mapTx.find(txHash);
+                if (it != mapTx.end()) {
+                    const CTransaction &txConflict = it->second.GetTx();
+                    if (txConflict != tx)
+                    {
+                        remove(txConflict, removedTxs, removedCerts, true);
+                    }
+                }
```

Comment:
I wonder if the `if (it != mapTx.end()) {` shouldn't be an assert. I mean you should be guaranteed that, if you find a nullifier in mempool, the tx containing it is in mempool as well right?
If so, either assert, or use directly 'mapTx.at(txHash)'

Reply:
I think that assert is not fully correct. See the logic above with `vin`.
For example, current Tx spends CSWs A and B and there is a conflict Tx' with the same CSWs A and B.
For CSW A we will find and remove Tx', and for B - will skip (because Tx' was removed before).

Reply:
I think the case for `vin` is different from current one, but I happy to discuss.
As per the "two cwss inputs" case, the `remove` over the first cws should correctly remove the containing tx from mapTx and mapSidechain too, hence processing of second cws would be skipped. I think it's important to add a UT confirming this

Reply:
Agreed AFK, add assert and leave a comment why.

Reply:
done